src/main/java/org/mockito/internal/creation/bytebuddy/InlineDelegateByteBuddyMockMaker.java

```diff
@@ -545,7 +549,11 @@ public void clearMock(Object mock) {
     @Override
     public void clearAllMocks() {
         mockedStatics.getBackingMap().clear();
-        mocks.clear();
```

Comment:
@raphw We introduced this API to clear up memory increases in continuous runs. I thought that these were the mocks in the backing map, but now I am starting to question myself. Do you think that if we merge this, we reintroduce these issues? Or was the memory leak in the handlers themselves?

Reply:
It looks like CyclicMockMethodArgumentMemoryLeakTest was [added](https://github.com/mockito/mockito/pull/1619) at the same time as clearAllMocks.  I confirmed that clearing the logic of clearAllMocks causes that test to fail with an OOM, and that the test still passes after this change; I believe that by replacing the mock records in the WeakHashMap, this change still allows all of the actual mock values to be GC'ed.

Does that match your memory?

Reply:
Ah yes, good point. That test looks correct to me, so good to confirm that it still passes as expected.